pkg/operator/observe_config.go

```diff
@@ -107,48 +136,59 @@ func (c ConfigObserver) sync() error {
 
 // observeEtcdEndpoints reads the etcd endpoints from the endpoints object and then manually pull out the hostnames to
 // get the etcd urls for our config. Setting them observed config causes the normal reconciliation loop to run
-func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, []error) {
 	etcdURLs := []string{}
 	etcdEndpoints, err := kubeClient.CoreV1().Endpoints(etcdNamespaceName).Get("etcd", metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		return observedConfig, nil
+		return observedConfig, []error{fmt.Errorf("endpoints/etcd.kube-system: not found")}
 	}
 	if err != nil {
-		return observedConfig, err
+		return observedConfig, []error{err}
+	}
+	dnsSuffix, ok := etcdEndpoints.Annotations["alpha.installer.openshift.io/dns-suffix"]
+	if !ok || dnsSuffix == "" {
```

Comment:
I think it should be okay to just check for the length of the string: `if len(dnsSuffix) == 0 {`

Reply:
Doesn't matter.

Reply:
> Doesn't matter.

no unnecesary `ok`